daliuge-translator/dlg/dropmake/web/main.js

```diff
@@ -23,84 +23,106 @@ $(document).ready(function () {
 
     $("#aboutModal #aboutLicense").load("/static/license.html")
     //keyboard shortcuts
-    var keyboardShortcuts = []
-    keyboardShortcuts.push({name:"Open Settings", shortcut:"O", code:79, action: "$('#settingsModal').modal('toggle')"})
-    keyboardShortcuts.push({name:"Deploy", shortcut:"D", code:75, action: "$('#shortcutsModal').modal('toggle')"})
-    keyboardShortcuts.push({name:"Open Keyboardshortcuts Modal", shortcut:"K", code:68, action: "$('#activeDeployMethodButton').click()"})
+    const keyboardShortcuts = [];
+    keyboardShortcuts.push({
+        name: "Open Settings",
+        shortcut: "O",
+        code: 79,
+        action: "$('#settingsModal').modal('toggle')"
+    })
+    keyboardShortcuts.push({
+        name: "Deploy",
+        shortcut: "D",
+        code: 75,
+        action: "$('#shortcutsModal').modal('toggle')"
+    })
+    keyboardShortcuts.push({
+        name: "Open Keyboardshortcuts Modal",
+        shortcut: "K",
+        code: 68,
+        action: "$('#activeDeployMethodButton').click()"
+    })
 
     //fill out keyboard shortcuts modal
-    keyboardShortcuts.forEach(element => { 
-        var shortCutItem =   '<div class="col-lg-6">'+
-                            '<div class="shortCutsModalItem">'+
-                                '<span>'+element.name+'</span>'+
-                                '<span class="shortCutsModalItemRight">'+element.shortcut+'</span>'+
-                            '</div>'+
-                        '</div>'
+    keyboardShortcuts.forEach(element => {
+        const shortCutItem = '<div class="col-lg-6">' +
+            '<div class="shortCutsModalItem">' +
+            '<span>' + element.name + '</span>' +
+            '<span class="shortCutsModalItemRight">' + element.shortcut + '</span>' +
+            '</div>' +
+            '</div>';
         $("#shortcutsModal .modal-body .row").append(shortCutItem)
     })
 
     //keyboard shortcuts execution
-    $(document).keydown(function(e){
-        if($("input").is(":focus")){
+    $(document).keydown(function (e) {
+        if ($("input").is(":focus")) {
             return
         }
-        keyboardShortcuts.forEach(element => { 
-            
-            if (e.which == element.code) //open settings modal on o
+        keyboardShortcuts.forEach(element => {
+
+            if (e.which === element.code) //open settings modal on o
             {
                 eval(element.action)
             }
         })
     })
 });
 
-function openSettingsModal(){
+function getCurrentPageUrl() {
+    const pathElements = window.location.href.split('/');
```

Comment:
**suggestion**: I think you should be able to get this stuff directly from `window.location.protocol` and `window.location.host`

Reply:
Yup, definitely reasonable.